presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedRegularFunctionId.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.function;
+
+import com.facebook.presto.spi.relation.FullyQualifiedName;
+import com.facebook.presto.spi.type.TypeSignature;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class SqlInvokedRegularFunctionId
+{
+    private final FullyQualifiedName name;
+    private final List<TypeSignature> argumentTypes;
+
+    @JsonCreator
+    public SqlInvokedRegularFunctionId(
```

Comment:
Why is this class necessary? Also why is this specific to Sql-invoked regular function? name + parameter types is the combination to resolve any functions not just for Sql-invoked regular functions. Also there's no need to require this to be part of the SqlInvokedRegularFunctionHandle. Each function namespace manager should have freedom in choosing how to identify a physical implementation of a function. Engine does not need to dictate what's in the handle.

Reply:
Removed for now. I will move it to spi when there are other users.